src/Microsoft.AspNet.JsonPatch/Converters/JsonPatchDocumentConverter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.JsonPatch.Exceptions;
+using Microsoft.AspNet.JsonPatch.Operations;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.AspNet.JsonPatch.Converters
+{
+    public class JsonPatchDocumentConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return true;
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, 
+            JsonSerializer serializer)
+        {
+            try
+            {
+                if (reader.TokenType == JsonToken.Null)
+                    return null;
+
+                // load jObject
+                var jObject = JArray.Load(reader);
+
+                // Create target object for Json => list of operations
+                var concreteList = typeof(List<Operation>);
+                var targetOperations = Activator.CreateInstance(concreteList);
+
+                // Create a new reader for this jObject, and set all properties 
+                // to match the original reader.
+                var jObjectReader = jObject.CreateReader();
+                jObjectReader.Culture = reader.Culture;
+                jObjectReader.DateParseHandling = reader.DateParseHandling;
+                jObjectReader.DateTimeZoneHandling = reader.DateTimeZoneHandling;
+                jObjectReader.FloatParseHandling = reader.FloatParseHandling;
+
+                // Populate the object properties
+                serializer.Populate(jObjectReader, targetOperations);
+
+                // container target: the JsonPatchDocument. 
+                var container = Activator.CreateInstance(objectType, targetOperations, new DefaultContractResolver());
```

Comment:
`new DefaultContractResolver()` isn't good, it means that we're starting with an empty cache.